### PR TITLE
Feature : OWASP RR support

### PR DESCRIPTION
**Pull request by syalioune**
### Description

Adding an OWASP RR calculator tab in vulnerability create/update modals. Few additions in display.

owasp-rr-dt-create
owasp-rr-dt
owasp-rr-dt-view-vuln

### Addressed Issue

https://github.com/DependencyTrack/dependency-track/issues/1493

### Additional Details

- Backend PR : https://github.com/DependencyTrack/dependency-track/pull/2275
- OWASP RR Calcultator PR : https://github.com/stevespringett/owasp-risk-rating-calculator/pull/54

### Checklist

- [x] I have read and understand the [contributing guidelines](https://github.com/DependencyTrack/dependency-track/blob/github-templates/CONTRIBUTING.md#pull-requests)
~~- [ ] This PR introduces new or alters existing behavior, and I have updated the [documentation](https://github.com/DependencyTrack/dependency-track/tree/master/docs/_docs) accordingly~~

### CODE REVIEW

File: src/views/portfolio/vulnerabilities/VulnerabilityCreateVulnerabilityModal.vue [hunk 1]
```diff
@@ -548,6 +842,19 @@ export default {
...
       return null;
     },
+    generateOwaspVector: function() {
+      if (this.owaspVector.sl != null && this.owaspVector.m != null && this.owaspVector.o != null
+        && this.owaspVector.s != null && this.owaspVector.ed != null
+        && this.owaspVector.ee != null &&this.owaspVector.a != null && this.owaspVector.id != null
+        && this.owaspVector.lc != null && this.owaspVector.li != null && this.owaspVector.lav != null
+        && this.owaspVector.lac != null && this.owaspVector.fd != null && this.owaspVector.rd != null
+        && this.owaspVector.nc != null && this.owaspVector.pv != null) {
+        let vector = `SL:${this.owaspVector.sl}/M:${this.owaspVector.m}/O:${this.owaspVector.o}/S:${this.owaspVector.s}/ED:${this.owaspVector.ed}/EE:${this.owaspVector.ee}/A:${this.owaspVector.sl}/ID:${this.owaspVector.id}/LC:${this.owaspVector.lc}/LI:${this.owaspVector.li}/LAV:${this.owaspVector.lav}/LAC:${this.owaspVector.lac}/FD:${this.owaspVector.fd}/RD:${this.owaspVector.rd}/NC:${this.owaspVector.nc}/PV:${this.owaspVector.pv}`;
```
Comment (nscuro) [inline src/views/portfolio/vulnerabilities/VulnerabilityCreateVulnerabilityModal.vue:852]: Skill Level is currently submitted as Awareness value.

```suggestion
        let vector = `SL:${this.owaspVector.sl}/M:${this.owaspVector.m}/O:${this.owaspVector.o}/S:${this.owaspVector.s}/ED:${this.owaspVector.ed}/EE:${this.owaspVector.ee}/A:${this.owaspVector.a}/ID:${this.owaspVector.id}/LC:${this.owaspVector.lc}/LI:${this.owaspVector.li}/LAV:${this.owaspVector.lav}/LAC:${this.owaspVector.lac}/FD:${this.owaspVector.fd}/RD:${this.owaspVector.rd}/NC:${this.owaspVector.nc}/PV:${this.owaspVector.pv}`;
```

File: src/views/portfolio/vulnerabilities/VulnerabilityCreateVulnerabilityModal.vue [hunk 2]
```diff
@@ -441,6 +625,106 @@ export default {
...
+          {text: this.$i18n.t('message.owasp_rr_threat_size_dev_sa'), value: '2'},
+          {text: this.$i18n.t('message.owasp_rr_threat_size_intranet'), value: '4'},
+          {text: this.$i18n.t('message.owasp_rr_threat_size_partners'), value: '5'},
+          {text: this.$i18n.t('message.owasp_rr_threat_size_auth_users'), value: '6'},
+          {text: this.$i18n.t('message.owasp_rr_threat_size_anonymous_internet_users'), value: '9'}
+        ],
+        lc: [
+          {text: this.$i18n.t('message.owasp_rr_loss_of_confidentiality_minimal_non_sensitive'), value: '2'},
+          {text: this.$i18n.t('message.owasp_rr_loss_of_confidentiality_minimal_critical'), value: '6'},
+          {text: this.$i18n.t('message.owasp_rr_loss_of_confidentiality_extensive_non_sensitive'), value: '6'},
```
Comment (nscuro) [inline src/views/portfolio/vulnerabilities/VulnerabilityCreateVulnerabilityModal.vue:658]: It's a bit odd to have multiple options map to the same factors. I thought it was a UI glitch at first 😅

image

Would it be possible to move those into a single option field?

Comment (syalioune) [reply src/views/portfolio/vulnerabilities/VulnerabilityCreateVulnerabilityModal.vue]: Done

File: src/views/portfolio/vulnerabilities/VulnerabilityCreateVulnerabilityModal.vue (hunk 1 above)
Comment (syalioune) [reply src/views/portfolio/vulnerabilities/VulnerabilityCreateVulnerabilityModal.vue]: Done

**Review by nscuro — APPROVED** [review]

**Pull request merged**